library/deletedrug.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/**
+ *  @package   OpenEMR
+ *  @link      http://www.open-emr.org
+ *  @author    Sherwin Gaddis <sherwingaddis@gmail.com>
+ *  @copyright Copyright (c )2020. Sherwin Gaddis <sherwingaddis@gmail.com>
+ *  @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ *
+ *
+ */
+
+require_once "../interface/globals.php";
+
+use OpenEMR\Common\Csrf\CsrfUtils;
+use OpenEMR\Common\Acl\AclMain;
+
+//ensure user has proper access
+if (!AclMain::aclCheckCore('admin', 'acl', 'super', 'patients')) {
```

Comment:
this doesn't look right. Would do:
```
if (!AclMain::aclCheckCore('patients', 'rx', '', 'write')) {
```

Reply:
or if you want only admin to do this, then would be:
```
if (!AclMain::aclCheckCore('admin',  'super')) {
```

Reply:
@bradymiller you are right, I forgot to remove patients. But really who do you think should have access. GMU did not specify. They wanted everyone that writes to be able to delete mistakes. What is your opinion?

Reply:
@juggernautsei ,
The natural route to go here then would be:
```
if (!AclMain::aclCheckCore('patients', 'rx', '', 'write')) {
```
(this is the same acl that controls being able to click the Add prescription button from the patient summary screen; so this would allow users that have permissions to add prescriptions to also delete them)

Reply:
This line still needs to be fixed. The one there now should break (it is not using the function correctly; check out examples in codebase for use). Rec changing to this:
```
if (!AclMain::aclCheckCore('patients', 'rx', '', 'write')) {
```

Reply:
hi @juggernautsei , This line still needs to be fixed (the line there now is not using the function correctly). See my posts above.